NumLib/ODESolver/NonlinearSolver-impl.h

```diff
@@ -146,21 +174,53 @@ solve(Vector &x)
 
         // std::cout << "  J:\n" << Eigen::MatrixXd(J) << std::endl;
 
-        if (!_linear_solver.solve(J, res, minus_delta_x)) {
+        bool iteration_succeeded = _linear_solver.solve(J, res, minus_delta_x);
+
+        if (!iteration_succeeded)
+        {
             ERR("The linear solver failed.");
-            BLAS::axpy(x, -_alpha, minus_delta_x);
-            success = false;
+        }
+        else
+        {
+            // TODO could be solved in a better way
```

Comment:
I suggest to change all TODO tags to \todo in a separate PR, and then all at once for all of OGS6, if that's important.